opencog/openpsi/OpenPsiImplicator.cc

```diff
@@ -127,3 +126,9 @@ Handle OpenPsiImplicator::imply(const Handle& rule)
     return Handle::UNDEFINED;
   }
 }
+
+OpenPsiImplicator& opencog::openpsi_implicator(AtomSpace* as)
+{
+  static OpenPsiImplicator implicator(as);
+  return implicator;
```

Comment:
I believe the issue will persist as long as you have statics like this.   The fix would be to code this as
```
OpenPsiImplicator& opencog::openpsi_implicator(AtomSpace* as)
{
   static OpenPsiImplicator implicator * = nullptr;
  if (nullptr ==  implicator) implicator = new OpenPsiImplicator(as);
  return *implicator; 
}
```
This will leak memory upon process exit, but it won't ever get a double free. 
You might want to double-check that this doesn't get called with multiple 
different atomspaces, as that might lead to trouble.  Some day in the future,
we will need to find a generic solution for this kind of thing, but that's for .. later.
The new to a static pointer will fix it.